metric-example/metric-expressions-for-k8s.md

```diff
@@ -0,0 +1,206 @@
+
+### Slack
+```
+(
+    (builtin:cloud.kubernetes.pod.cpuRequests:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+    -(builtin:containers.cpu.usageMilliCores:parents:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+)
+:splitBy("dt.entity.cloud_application"):avg
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+*Note: We are using **:parents**, to be able to split and filter by entities being higher in the Kubernetes entity hierarchy, which is defined as follows: container -> pod -> workload -> namespace -> cluster.
+You can also increase the scope of this query to multiple workloads, by expanding the scope of the filter or just removing it. Just be aware, that in very large environments, this can put a lot of stress onto your Dynatrace environment.
+
+### Usage above requests
+Using the following query, we can see which workloads have a higher CPU usgae, than requested. Meaning, for such workloads one should consider increasing the set requests to increase the stabilitiy of this workload as well as the stabilitiy of the Kubernetes cluster it is running on.
+```
+(
+    (builtin:containers.cpu.usageMilliCores:parents:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+    -(builtin:cloud.kubernetes.pod.cpuRequests:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):avg)
+)
+:splitBy("dt.entity.cloud_application"):avg
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+
+### Usage in terms of limits
+Users afraid of throttling, often want to alert on the percentage of limits being used. However, we suggest to alert on throttling relative to usage. Reason for this is, that the relative usage in terms of limits is not the only deciding factor for when Kubernetes starts to throttle containers - we'll cover this in the next example.
+```
+(
+    (builtin:containers.cpu.usageMilliCores:avg:parents:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):sum)
+    /(builtin:cloud.kubernetes.pod.cpuLimits:avg:parents:splitBy("dt.entity.cloud_application_instance","dt.entity.cloud_application"):sum)
+    *(100)
+)
+:splitBy("dt.entity.cloud_application"):avg
+:filter(and(in("dt.entity.cloud_application",entitySelector("type(cloud_application),entityId(~"CLOUD_APPLICATION-A26E32FC302257AB~")"))))
+```
+
+### High trotthling
+As mentioned, it makes more sense to alert on the outcome rather than only on one of mulitple potential triggers for throttling. In other words, track throttling relative to usage instead of usage in terms of limits.
```

Comment:
```suggestion
As mentioned, it makes more sense to alert on the outcome rather than only on one of multiple potential triggers for throttling. In other words, track throttling relative to usage instead of usage in terms of limits.
```